### ISSUE TITLE
Add support for bulk addition of points

### ISSUE THREAD

**Issue by mjulian** [ripienaar/travlrmap#39]
It would be very useful to have support for bulk addition of points, through either pasting in a list of locations to a text field, or uploading a list in a specific format.

**Comment by ripienaar** [comment]
when I needed that it seemed the same amount of work to make a yaml file as it would be to make say a csv file or other format?  Do you have a suggestion for a format?

**Comment by mjulian** [comment]
Thinking from the user perspective, and use cases:
- A user has a list of places they've visited and wants to get them into a travlrmap instance. Their list is most likely in Excel (easily exported to CSV) or a simple text file.
- A user finds travlrmap and decides to use it, spending the next few minutes making a list of all the places they've been, most likely in a simple text file.

Based on those, I think requiring the user to format the list into YAML first would be too burdensome. Accepting a CSV/TXT file or a text box entry field would be the best, in my opinion.

The only use case I can think of for accepting formats other than CSV/TXT would be importing from Google Earth or some such (GeoRSS, KML, etc), which is maybe a lot of work for not much user benefit? I don't know there, just guessing.

**Comment by ripienaar** [comment]
Yeah sounds good - geocoding is not that awesome and often get it wrong in my experience so will need a UI For a user to review them all.

And the thing I fear most - and why i held off letting the geocoder write data - is I'd need to make an edit screen for all the points etcetc which I doubt I have time for.

It's a good idea though, will keep it up here who knows maybe I have a bored weekend soon :)

**Comment by mjulian** [comment]
You could attack the 80% of the problem: let the geocoder take the first result, put it on the preview map, and do that for all of them. For any that are too ambiguous to locate, kick back an error above the map ("The following locations were unsuccessful:"). Skip the editing requirement, and put that back on the user to provide detailed enough locations (eg, 'London, UK' vs 'London'). The user then has a map they can look at to see if anything appears wrong, and update the input accordingly.

**Comment by ripienaar** [comment]
Yeah, roughly what I had in mind - Google APIs will freak out about lots of geocoding requests though so rapidly :) So maybe consume the list, fill in a table and have a button next to each to do the lookup which would popup a little map with the geocoded result and a confirm box?

Anyway, I'll take a stab at it, doesn't sound too bad - am traveling most of feb though so wont be soon.

**Comment by mjulian** [comment]
That sounds like a pretty good approach to me. No rush, of course, just thought it'd be a good idea. :)
